test/UnitTests/Microsoft.Testing.Platform.UnitTests/Helpers/TaskExtensionsTests.cs

```diff
@@ -72,49 +72,47 @@ public async Task CancellationAsync_NonCanceledWithArgument_Succeeds()
 
     [TestMethod]
     public async Task CancellationAsync_ObserveException_Succeeds()
-        => await RetryHelper.RetryAsync(
-            async () =>
-            {
-                ManualResetEvent waitException = new(false);
-                await Assert.ThrowsAsync<OperationCanceledException>(async ()
-                    => await Task.Run(
-                        async () =>
-                        {
-                            await Task.Delay(TimeSpan.FromSeconds(10), TestContext.CancellationToken);
-                            waitException.Set();
-                            throw new InvalidOperationException();
-                        }, TestContext.CancellationToken).WithCancellationAsync(new CancellationTokenSource(TimeSpan.FromSeconds(1)).Token));
-
-                waitException.WaitOne();
-                await Task.Delay(TimeSpan.FromSeconds(4), TestContext.CancellationToken);
-            }, 3, TimeSpan.FromSeconds(3));
+    {
+        ManualResetEvent waitException = new(false);
+        CancellationToken token = new CancellationTokenSource(TimeSpan.FromSeconds(1)).Token;
+        OperationCanceledException ex = await Assert.ThrowsAsync<OperationCanceledException>(async ()
+            => await Task.Run(
+                async () =>
+                {
+                    await Task.Delay(TimeSpan.FromSeconds(5), TestContext.CancellationToken);
+                    waitException.Set();
+                    throw new InvalidOperationException();
+                }, TestContext.CancellationToken).WithCancellationAsync(token));
+#if !NETFRAMEWORK // Polyfill bug in Task.WhenAsync implementation :/
+        Assert.AreEqual(token, ex.CancellationToken);
+#endif
+        waitException.WaitOne();
+    }
 
     [TestMethod]
     public async Task CancellationAsyncWithReturnValue_ObserveException_Succeeds()
-        => await RetryHelper.RetryAsync(
-            async () =>
+    {
+        ManualResetEvent waitException = new(false);
+        CancellationToken token = new CancellationTokenSource(TimeSpan.FromSeconds(1)).Token;
+        OperationCanceledException ex = await Assert.ThrowsAsync<OperationCanceledException>(async ()
+            => await Task.Run(async () =>
```

Comment:
can we elide the asyncs here?

Reply:
I don't think it brings much value so I prefer to keep the test as-is.